geode-protobuf/src/test/java/org/apache/geode/internal/protocol/protobuf/v1/operations/GetAllRequestOperationHandlerJUnitTest.java

```diff
@@ -69,6 +74,34 @@ public void setUp() throws Exception {
     operationHandler = new GetAllRequestOperationHandler();
   }
 
+  @Test
+  public void processReturnsErrorUnableToDecodeRequest() throws Exception {
+    Exception exception = new DecodingException("error finding codec for type");
+    ProtobufSerializationService serializationServiceStub =
+        mock(ProtobufSerializationService.class);
+    when(serializationServiceStub.decode(any())).thenThrow(exception);
+
+    BasicTypes.EncodedValue encodedKey = BasicTypes.EncodedValue.newBuilder()
+        .setJsonObjectResult("{\"someKey\":\"someValue\"}").build();
+    Set<BasicTypes.EncodedValue> keys = Collections.<BasicTypes.EncodedValue>singleton(encodedKey);
+    RegionAPI.GetAllRequest getRequest =
+        ProtobufRequestUtilities.createGetAllRequest(TEST_REGION, keys);
+
+    Result response = operationHandler.process(serializationServiceStub, getRequest,
+        TestExecutionContext.getNoAuthCacheExecutionContext(cacheStub));
+
+    assertTrue("response was " + response, response instanceof Success);
+
+    RegionAPI.GetAllResponse message = (RegionAPI.GetAllResponse) response.getMessage();
+    assertEquals(1, message.getFailuresCount());
+
+    BasicTypes.KeyedError error = message.getFailures(0);
+    assertEquals(BasicTypes.ErrorCode.INVALID_REQUEST, error.getError().getErrorCode());
+    assertTrue(error.getError().getMessage().contains("encoding not supported"));
```

Comment:
Not really a fan of asserting specific error message verbiage, but that's just a minor quibble.